Extension/src/LanguageServer/client.ts

```diff
@@ -1508,15 +1508,19 @@ export class DefaultClient implements Client {
                 if (!config) {
                     return;
                 }
-                // TODO: This is a hack to get around CMake Tools bug: https://github.com/microsoft/vscode-cmake-tools/issues/1073
                 let foundMatch: boolean = false;
-                for (let c of config.browsePath) {
-                    if (vscode.workspace.getWorkspaceFolder(vscode.Uri.file(c)) === this.RootFolder) {
-                        foundMatch = true;
-                        break;
+                if (currentProvider.version < Version.v3) {
+                    // This is to get around the (fixed) CMake Tools bug: https://github.com/microsoft/vscode-cmake-tools/issues/1073
+                    for (let c of config.browsePath) {
+                        if (vscode.workspace.getWorkspaceFolder(vscode.Uri.file(c)) === this.RootFolder) {
+                            foundMatch = true;
+                            break;
+                        }
                     }
-                }
-                if (foundMatch) {
+                    if (foundMatch) {
+                        this.sendCustomBrowseConfiguration(config, currentProvider.extensionId);
```

Comment:
replace line 1516 with 1521 and remove `foundMatch` flag?

Reply:
Yeah, good catch.